src/Tribe/Shortcode/Event_Details.php

```diff
@@ -0,0 +1,214 @@
+<?php
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+/**
+ * Setup the Event Details Shortcode to be able to place the Details for a Event on other pages
+ *
+ * @since 4.1
+ */
+class Tribe__Events__Shortcode__Event_Details {
+
+	/**
+	 * We use a Private constructor to Prevent `new Class` usage
+	 * Correct usage: `Tribe__Events__Shortcode__Event_Details::instance()`
+	 *
+	 * @since  4.1
+	 * @return  void
+	 */
+	private function __construct() {
```

Comment:
Totally optional, just me nitpicking: what about collapsing the method body down to `{}` (since we're only declaring the constructor to control its visibility, no point in having it occupy extra lines).